buildSrc/src/main/kotlin/tasks/InstallAndroidSdkTask.kt

```diff
@@ -0,0 +1,89 @@
+package tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.UntrackedTask
+import org.gradle.tooling.GradleConnector
+import java.io.File
+
+/**
+ * Installs Android SDK. This task requires the ANDROID_SDK_ROOT env variable to be set.
+ */
+@UntrackedTask(because = "Output directory can change when running other builds")
+abstract class InstallAndroidSdkTask : DefaultTask() {
+
+    @get:Input
+    abstract val androidSdkVersion: Property<String>
+
+    @get:OutputDirectory
+    abstract val androidProjectDir: DirectoryProperty
+
+    @TaskAction
+    fun install() {
+        if (System.getenv("ANDROID_SDK_ROOT") == null) {
+            throw GradleException(
+                "ANDROID_SDK_ROOT env variable is not set but it should be with the installation directory path. " +
+                    "The installation directory should also contain the Android sdk license key. See https://developer.android.com/studio/intro/update.html#download-with-gradle."
+            )
+        }
+
+        val androidSdkVersion = androidSdkVersion.get()
+        val projectDir = androidProjectDir.get().asFile
+
+        createAndroidProject(projectDir, androidSdkVersion)
+        buildAndroidProject(projectDir)
+    }
+
+    private fun buildAndroidProject(projectDir: File) {
+        val connector = GradleConnector.newConnector()
+            .forProjectDirectory(projectDir)
+        connector.connect().use {
+            it.newBuild().forTasks("build").run()
```

Comment:
Ah, one more thing: I think it would be good to print out the output of the build when it fails, WDYT?

Reply:
Added. I had to handle it manually by catching exception. Previously it worked if you used --stacktrace, but now it's always visible